### PR TITLE
Lesson Builder Transition, Part 1/???

### LINKED ISSUE #421: Create Lesson Builder submenu

### LINKED ISSUE #422: Build Lesson first resource image view

### LINKED ISSUE #423: Implement "student version" button
A “View Student Version” button will link to the “student version” of the lesson plan which will omit the Educator Notes field.

### LINKED ISSUE #425: Create lesson related resources view
All related resources will be displayed in a numbered list featuring the resource title, author name, and media type.

### LINKED ISSUE #426: Create lesson questions view
All questions will be displayed in a numbered list.

### LINKED ISSUE #427: Configure lesson metatags
- [x] title
- [x] description
- [x] og:title
- [x] og:type
- [x] og:url
- [x] og:image
- [ ] og:video
- [x] twitter:card (Twitter card type)
- [x] twitter:creator
- [x] twitter:url
- [x] twitter:title

### LINKED ISSUE #428: Configure lesson path alias pattern

### LINKED ISSUE #430: Create lesson context
Mainly to populate sidebar:
>A “View Student Version” button will link to the “student version” of the lesson plan which will omit the Educator Notes field.
>
>A “Print this Lesson Plan” button will launch the print dialog. A minimal print stylesheet will be preserved.
>
>All related resources will be displayed in a numbered list featuring the resource title, author name, and media type.
>
>All questions will be displayed in a numbered list.

### LINKED ISSUE #431: Create lesson builder landing page view
featured first, then chronological order

exposed filters:
- Keywords (full-text search)
- Published date (start-end range)
- Issue
- Tag
- Region
- Country
- Author
- Subject
- Grade level

### LINKED ISSUE #432: Configure/template lesson teaser view mode
Each lesson plan teaser item will include:
- Image pulled from the first related resource, if available
- “Featured” badge
- Lesson title
- Author name
- Grade level
- Teaser text

### LINKED ISSUE #433: Create Lesson Plan Grouping vocabulary

### LINKED ISSUE #434: Add Lesson Plan Grouping field to lessons

### LINKED ISSUE #435: Create lesson plan grouping term page view

### LINKED ISSUE #436: Remove ng app from /builder

### LINKED ISSUE #438: Delete deprecated fields
The “Finished” and “Model Lessons” fields on lessons will be deleted.

### LINKED ISSUE #439: Export list of lesson builder users
Include:

- [x] email address

- [x] first & last name (where available)

- [x] title (where available)

- [x] URLs of authored Lesson Plans

### LINKED ISSUE #440: Delete draft and orphaned lessons

### LINKED ISSUE #442: Create lesson builder section context

### LINKED ISSUE #444: Configure lesson default view mode

### PR DESCRIPTION
**Pull request by xaqrox**
http://pccr.radcampaign.com

To test locally, pull, updb, revert.

After all that (and on the AWS instance), the following should be true:

### Landing Page
- [x] `/builder` is a Drupal view, not an AngularJS app (close #436, close #431)
- [x] Each lesson in on the landing page displays the "Featured Image" if available, or the first available image from its resources, and a ":star: Featured" badge if appropriate. (close #432)
- [x] Submenu with "Lesson Builder" heading and "Browse Lessons" link appears on every path that starts with `/builder` (close #421, close #442)

### Lessons
- [x] All remaining lessons are published and have non-anonymous authors (close #440)
- [x] The "Finished" and "Model Lesson" fields are gone from Lessons (close #438 )
- [x] New and existing lessons get/have aliases of the pattern `builder/lesson/[node:title]-[node:nid]`. There are still duplicate titles, thought node ids looked more official than `-0`, `-1`, etc. (close #428, close #423)
- [x] Lessons have an image view, educator notes, and full body content in the main content area (close #444, close #422)
- [x] "View Student Version" button, "Print This Lesson" button, resource list, and questions appear in the sidebar of lessons (close #430)
- [x] Clicking the "View Student Version" button adds `?student=1` to the URL and hides the Educator notes (close #423)
- [x] Clicking the "Print this Lesson Plan" button brings up the print preview dialog (not that you'd want to print what comes up yet) (#424)
- [x] Resource list features resource title, media type, and author of each resource (close #425)
- [x] Questions appear in sidebar with whatever formatting the user entered (close #426)
- [x]  All the metatags needed to resolve #427 are present on lessons

### Lesson Plan Groupings
- [x]  Lessons have an autocomplete Lesson Plan Grouping field, and term pages (`/admin/structure/taxonomy/lesson_grouping`) display related lessons (close #433, close #434, close #435)

### Users
- [x] CSV of all lesson builder users available at `/lesson-builder-users/export.csv` (close #439)
- [x] Anonymous users can not create visitor accounts (#441)

:boom:

### CODE REVIEW

**Comment by jselt** [comment]
Featured tag needs more left position.

featuredpng


 also there isn't enough contrast between white text on a yellow background to meet accessibility standards. - http://webaim.org/resources/contrastchecker/

**Comment by jselt** [comment]
Needs spacing between rows

rows

**Comment by jselt** [comment]
Lesson Builder menu hover state needs better contrast - fails again - http://webaim.org/resources/contrastchecker/

**Comment by jselt** [comment]
Search Form (Advanced):

Post Date is a weird label - maybe "Lesson Date"? 

Also, it doesn't bring up a date picker, that would be ok but it then at least needs some hints for the user as to what format to enter (placeholder text at a minimum.

**Comment by daniellerad** [comment]
I'd also mention that maybe it's best to be consistent with the dates configuration on the Reporting page filters? http://pccr.radcampaign.com/reporting

**Comment by jselt** [comment]
http://pccr.radcampaign.com/builder/lesson/what-happens-communities-when-industries-leave-21230

Lot of spacing between Student Version Buttons and Print Buttons - reduce. Ideally shorten button text and get side by side at least at largest breakpoint.

**Comment by jselt** [comment]
No way to get back from Student Version - not sure if this is entirely necessary - but hard to rely on everyone to know how to remove the url param.

**Comment by jselt** [comment]
Spacing in notes:
Needs some.
notes

**Comment by jselt** [comment]
Url color on Builder node bodies/fields - needs to override the default link colors as in Stories:

http://pccr.radcampaign.com/reporting/syrian-refugees-living-jordan-journalism-offers-hope-and-opportunity-rebuild-their

**Comment by jselt** [comment]
Builder tag page - http://pccr.radcampaign.com/builder-grade-level/middle-school

"Lessons" tab looks really odd on it's own:

middle school pulitzer center

Not sure what the other tabs are supposed to be - but maybe when the count is one we need a diff style here?

**Comment by xaqrox** [comment]
### In last commit (7d24593):
- featured badge positioned better
- featured badge uses same colors as event ongoing badge
- row spacing on landing page
- better lb submenu hover color
- chaged post date label, added end date, datepicker (forgot to export before)
- more line height for educator notes field (match body field)
- generalized body/wysiwyg field height to apply to educator notes

### Questions:
- Should the Print and Student Version be somewhere else on the page? After Resources? Between body text and educator notes? I'm gonna put them into the same block to get around the standard block `margin-bottom`
- @daniellerad thoughts on if we need an obvious way back from student version?
- The single-tab situation exists all over the site. For example: http://pccr.radcampaign.com/publications/milwaukee-journal-sentinel It does look awkward. Just hide it?

**Comment by daniellerad** [comment]
Should the Print and Student Version be somewhere else on the page? After Resources? Between body text and educator notes? I'm gonna put them into the same block to get around the standard block margin-bottom
- I like the buttons where they are. Just need a bit less space around them.

Thoughts on if we need an obvious way back from student version?
- I think Jeff specifically does not want that, but I understand the user experience issues. Let's hold off and I can prompt him.

The single-tab situation exists all over the site. For example: http://pccr.radcampaign.com/publications/milwaukee-journal-sentinel It does look awkward. Just hide it?
- If it's easy. Otherwise, leave it. The client is aware of it and fine with it.

**Comment by jselt** [comment]
Looks good - I agree with @daniellerad about the two buttons.

File: sites/all/modules/custom/lesson_builder/lesson_builder.install [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    drupal_uninstall_modules([$name]);
+  }
+}
+
+/**
+ * Add lesson builder menu links.
+ */
+function lesson_builder_update_7007(&$sandbox) {
+  // Publish webform nodes.
+  $webform_info = [
```
Comment (xaqrox) [inline sites/all/modules/custom/lesson_builder/lesson_builder.install]: @jselt here are the nids for the webforms and the signup page

**Pull request merged**